### PR TITLE
follow up on `DataCombined`

### PR DESCRIPTION
**Pull request by IndrajeetPatil**
- [x] #735
- [x] #743 
- [x] #727
- [x] #728  
- [x] refactoring and best practices

### CODE REVIEW

File: tests/testthat/test-data-combined.R [hunk 1]
```diff
@@ -117,46 +115,103 @@ test_that("dataCombined - either dataSet or SimulationResults provided", {
...
+  myCombDat$addDataSets(dataSet[[1]])
+  myCombDat$setGroups(list("Stevens_2012_placebo.Placebo_total" = "m"))
+
+  expect_equal(myCombDat$groupMap$group, "m")
+
+  myCombDat$setGroups(list("Stevens_2012_placebo.Placebo_total" = NA))
+  expect_equal(myCombDat$groupMap$group, NA_character_)
+})
+
+test_that("setting groups with atomic vector or a list shouldn't make a difference", {
```
Comment (msevestre) [inline tests/testthat/test-data-combined.R:153]: This is so much easier to discover the API. Well done @IndrajeetPatil

**Review by msevestre — APPROVED** [review]
Looks good to me. I have to say that I find this new API much easier to comprehend. Well done.
Also the tests refactoring is a pure joy.

**Comment by IndrajeetPatil** [comment]
Thanks! 😸 

brubalL

**Comment by PavelBal** [comment]
@msevestre I actually wanted to review, but...

**Comment by msevestre** [comment]
it's not too late: )

I want to close all PR because we need to fix this files issue stuff. For that we need to reconcile all files with the expected EOL

**Comment by msevestre** [comment]
so please review and we can create issues for what needs to be changed Sorry about this @PavelBal

**Comment by PavelBal** [comment]
Ok no problem, thanks.

**Pull request merged**